xmtp_mls/src/groups/tests/test_dm.rs

```diff
@@ -0,0 +1,30 @@
+use xmtp_db::consent_record::{ConsentState, ConsentType};
+
+use crate::tester;
+
+/// Test case: If two users are talking in a DM, and one user
+/// creates a new installation and creates a new DM before being
+/// welcomed into the old DM, that new DM group should be consented.
+#[xmtp_common::test(unwrap_try = "true")]
+async fn auto_consent_dms_for_new_installations() {
+    tester!(alix);
+    tester!(bo1);
+    // Alix and bo are talking fine in a DM
+    alix.test_talk_in_dm_with(&bo1).await?;
+
+    tester!(bo2, from: bo1);
+
+    // Bo creates a new installation and immediately creates a new DM with alix
+    let bo2_dm = bo2
+        .find_or_create_dm_by_inbox_id(alix.inbox_id(), None)
+        .await?;
+
+    // Alix pulls down the new DM from bo
+    alix.sync_welcomes(&alix.provider).await?;
+
+    // That DM should be already consented, since alix consented with bo in another DM
+    let consent = alix
+        .get_consent_state(ConsentType::ConversationId, hex::encode(bo2_dm.group_id))
+        .await?;
+    assert_eq!(consent, ConsentState::Allowed);
```

Comment:
This asserted to `Unknown` before this change.

Reply:
Awesome love the test 🔥